specification/search/Azure.Search/models-service.tsp

```diff
@@ -0,0 +1,9050 @@
+  /**
+   * The similarity metric to use for vector comparisons.
+   */
+  metric?: VectorSearchAlgorithmMetric;
+}
+
+/**
+ * Contains configuration options specific to the scalar quantization compression
+ * method used during indexing and querying.
+ */
+model ScalarQuantizationCompression extends VectorSearchCompression {
+  /**
+   * Contains the parameters specific to Scalar Quantization.
+   */
+  scalarQuantizationParameters?: ScalarQuantizationParameters;
+
+  /**
+   * The name of the kind of compression method being configured for use with vector
+   * search.
+   */
+  kind: "scalarQuantization";
+}
+
+/**
+ * Contains the parameters specific to Scalar Quantization.
+ */
+model ScalarQuantizationParameters {
+  /**
+   * The quantized data type of compressed vector values.
+   */
+  quantizedDataType?: VectorSearchCompressionTarget;
+}
+
+/**
+ * Contains configuration options specific to the binary quantization compression
+ * method used during indexing and querying.
+ */
+model BinaryQuantizationCompression extends VectorSearchCompression {
+  /**
+   * The name of the kind of compression method being configured for use with vector
+   * search.
+   */
+  kind: "binaryQuantization";
+}
+
+/**
+ * Specifies the Azure OpenAI resource used to vectorize a query string.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Maintain compatibility with existing models."
+model AzureOpenAIVectorizer extends VectorSearchVectorizer {
+  /**
+   * Contains the parameters specific to Azure OpenAI embedding vectorization.
+   */
+  azureOpenAIParameters?: AzureOpenAIVectorizerParameters;
+
+  /**
+   * The name of the kind of vectorization method being configured for use with
+   * vector search.
+   */
+  kind: "azureOpenAI";
+}
+
+/**
+ * Specifies the parameters for connecting to the Azure OpenAI resource.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Maintain compatibility with existing models."
+model AzureOpenAIVectorizerParameters {
+  /**
+   * The resource URI of the Azure OpenAI resource.
+   */
+  resourceUri?: url;
+
+  /**
+   * ID of the Azure OpenAI model deployment on the designated resource.
+   */
+  deploymentId?: string;
+
+  /**
+   * API key of the designated Azure OpenAI resource.
+   */
+  apiKey?: string;
+
+  /**
+   * The user-assigned managed identity used for outbound connections.
+   */
+  authIdentity?: SearchIndexerDataIdentity;
+
+  /**
+   * The name of the embedding model that is deployed at the provided deploymentId
+   * path.
+   */
+  modelName?: AzureOpenAIModelName;
+}
+
+/**
+ * Abstract base type for data identities.
+ */
+@discriminator("@odata.type")
+model SearchIndexerDataIdentity {
+  /**
+   * A URI fragment specifying the type of identity.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "Existing"
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "Maintain compatibility with existing models."
+  `@odata.type`: string;
```

Comment:
Here for example it looks like we should have an enum with 2 values that correspond to those that are specified in the `SearchIndexerDataNoneIdentity` and `SearchIndexerDataUserAssignedIdentity` models. Then you use those enum values to specify the discriminator value on the child models. Even if the swagger didnt specify this, it would be something we should fix in the tsp migration since tsp catches more of these issues that slipped through with handwritten swaggers